Add initial compilation

znicholls opened the pull request:
@danielhuppmann this should be enough to get us started. It's basically just a big dump of data but I think gives us a base we can then improve on. 

I've made issues for everything I can think of that is missing (really only using the nomenclature package as that also handles units checking), see #2

znicholls committed:
Add basic setup

znicholls committed:
Remove test of generation from raw data

znicholls committed:
Add tests of badc reader

znicholls committed:
Format

znicholls committed:
Add figure SPM.1 data

znicholls committed:
Add Fig SPM.8

znicholls committed:
Fig SPM.4

znicholls committed:
Fig SPM.10

znicholls commented:
(p.s. I will make PRs from forks in future)

znicholls committed:
Format

znicholls committed:
Update setup.cfg

znicholls committed:
Add processed data

znicholls committed:
Add data README

znicholls committed:
Fix up leftover comments

danielhuppmann commented on README.md:
```diff
@@ -30,8 +41,5 @@ and re-use in subsequent research and analysis.
-   (note that this package is distributed on pypi as `pyam-iamc`)
- - [scmdata](https://scmdata.readthedocs.io) 
+The data and notebooks in this repository use the dependencies specified in `environment.yml`.
+These can be installed with conda: `conda env install -f environment.yml`.
```
Doesn't work for me, maybe `conda env create -f environment.yml` instead?

znicholls replied:
Oops yes very lazy from me

danielhuppmann commented on data/README.md:
```diff
@@ -0,0 +1,12 @@
+The `processed` directory contains the processed data products, also as created by the compilation scripts (see `README.md` in the root of this repository for instructions about compiling the data).
+All data in this directory is in a uniform csv-based format, ready for use directly with automated processing software and notably perfectly suited for use with the scmdata and pyam repositories.
+
+The `dummy` directory contains data in the format of data products, but where actual data as used in the AR6 WG1 is not yet available.
```
This was for testing some routines before WG1 data was available, I would not include it in this repo.

znicholls replied:
Alles klar

danielhuppmann commented on README.md:
```diff
@@ -30,8 +41,5 @@ and re-use in subsequent research and analysis.
- - [pyam](https://pyam-iamc.readthedocs.io)
-   (note that this package is distributed on pypi as `pyam-iamc`)
- - [scmdata](https://scmdata.readthedocs.io) 
+The data and notebooks in this repository use the dependencies specified in `environment.yml`.
```
I would actually keep both the high-level overview of the key packages that are required, and then add the technical instructions on how to install it...

znicholls replied:
Ok nice will put back in

danielhuppmann commented on README.md:
```diff
@@ -12,6 +12,17 @@ The data is available at the [CEDA Archive](https://data.ceda.ac.uk/badc/ar6_wg1
 under a Creative Commons CC-BY license, but the formats are quite diverse
 and not easy to handle.
 
+### Compiling the data
```
My idea was to first have a high-level overview, then dive into the weeds - so maybe move this out of "Overview" and into an own section later?

znicholls replied:
👍

danielhuppmann commented on data/raw/dap.ceda.ac.uk/README.md:
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+OUTPUT_DIRECTORY="./"
+
+wget -e robots=off --mirror --no-parent -r https://dap.ceda.ac.uk/badc/ar6_wg1/data/spm/spm_01/v20210809/ --directory-prefix="${OUTPUT_DIRECTORY}"
```
If this is taken from the `config.yaml`, wouldn't it make more sense to refer to that file? Otherwise, we risk getting out of sync between the explicit list and the config...

znicholls replied:
Ok I made it more explicit. I agree it's a risk of getting out of sync. I think the only way to properly mitigate that risk would be to add CI so I made #4.

danielhuppmann reviewed (commented):
Looks good as a start, a few comments inline, and suggestion to use the following as a header for all notebooks:

```markdown
# AR6 WG1 - SPM.x

This notebook reproduceds parts of **Figure SPM.x** of the IPCC's *Working Group I contribution to the Sixth Assessment Report* ([AR6 WG1](https://www.ipcc.ch/assessment-report/ar6/)).

The data supporting the SPM figure is published under a Creative Commons CC-BY license at 
the [Centre for Environmental Data Analyis (CEDA)](https://catalogue.ceda.ac.uk/uuid/ae4f1eb6fce24adcb92ddca1a7838a5c).

This notebook uses a version of that data which was processed for interoperability with the format used by IPCC WG3, the so-called IAMC format.  
The notebook is available under an open-source [BSD-3 License](https://github.com/openscm/AR6-WG1-Data-Compilation/blob/main/LICENSE) in the [openscm/AR6-WG1-Data-Compilation](https://github.com/openscm/AR6-WG1-Data-Compilation) GitHub repository.

The notebook uses the community Python package [scmdata|(https://scmdata.readthedocs.io), <a short blurb about the package>.
```

znicholls committed:
Fix conda install instruction

znicholls committed:
Remove dummy dat directory

znicholls committed:
Put requirements back in README

znicholls committed:
Moved data compilation to its own section in README

znicholls committed:
Make download instructions relative to config yaml

znicholls committed:
Cleanup notebooks headers and naming

znicholls commented:
> suggestion to use the following as a header for all notebooks

Nice, updated the notebooks and updated the naming based on our offline discussion.

danielhuppmann reviewed (approved):
Looks great, many thanks! The link to scmdata in the notebooks is formatted incorrectly - please fix, then good to be merged!

znicholls commented:
> The link to scmdata in the notebooks is formatted incorrectly

fml

znicholls committed:
Fix formatting in notebook
